src/javascript/app/pages/trade/purchase/purchase.jsx

```diff
@@ -0,0 +1,337 @@
+                                                size='lg'
+                                                label={localize('Top up')}
+                                                onClick={() => {
+                                                    data?.error?.action?.();
+                                                    hidePurchaseResults();
+                                                }}
+                                            />
+                                        </>
+                                    )}
+                                </>
+                            ) : (
+                                <span className='error-message'>
+                                    <Text size='md' centered>{data?.error.message}</Text>
+                                </span>
+                            )}
+                        </div>
+                    </div>
+                </div>
+            </div>);
+    }
+
+    return (
+        <div className='quill-purchase-section-results'>
+            <div className='top-box'>
+                <div className='header-box'>
+                    <Text size='xl'>{data?.pr_heading}</Text>
+                    <span
+                        className='close-btn'
+                        onClick={() => {
+                            triggerClick('#close_confirmation_container');
+                            hidePurchaseResults();
+                        }}
+                    >
+                        <LabelPairedXmarkMdRegularIcon />
+                    </span>
+                </div>
+              
+                <div className='body-box'>
+                    <div className='description-box'>
+                        <Text size='sm'>{data?.pr_description}</Text>
+                    </div>
+                    <div className='table-box'>
+                        {data?.pr_tablePayoutValue && (
+                            <div className='table-item'>
+                                <Text size='sm' centered bold>{data?.pr_tablePayout}</Text>
+                                <Text size='sm' centered >{parseData(data.pr_tablePayoutValue)}</Text>
+                            </div>
+                        )}
+                        { data?.pr_tableCostValue && (
+                            <div className='table-item'>
+                                <Text size='sm' centered bold>{data?.pr_tableCost}</Text>
+                                <Text size='sm' centered >{parseData(data.pr_tableCostValue)}</Text>
+                            </div>
+                        )}
+                        { data?.pr_tableProfitValue && data?.pr_showTableProfit && (
+                            <div className='table-item'>
+                                <Text size='sm' centered bold>{data?.pr_tableProfit}</Text>
+                                <Text size='sm' centered>{parseData(data.pr_tableProfitValue)}</Text>
+                            </div>
+                        )}
```

Comment:
Some blocks of `Text`'s and, for example, `<div className='purchase-amount-info-box'>` blocks above are used repeatedly, they could be mapped through based on data/content :)
Mapping of similar blocks would help reduce this huge jsx